contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -78,6 +98,29 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
     }
   }
 
+  /// @notice Updates a node operator
+  /// @param nodeOperatorIds The ID of the node operator being updated
+  function updateNodeOperators(uint256[] calldata nodeOperatorIds, NodeOperator[] calldata nodeOperators) external {
+    if (nodeOperatorIds.length != nodeOperators.length)
+      revert LengthMismatch(nodeOperatorIds.length, nodeOperators.length);
+    for (uint256 i; i < nodeOperatorIds.length; ++i) {
+      uint256 nodeOperatorId = nodeOperatorIds[i];
+      NodeOperator memory nodeOperator = nodeOperators[i];
+      if (nodeOperator.admin == address(0)) revert InvalidNodeOperatorAdmin();
+      if (msg.sender != nodeOperator.admin && msg.sender != owner()) revert AccessForbidden();
+
+      if (
+        s_nodeOperators[nodeOperatorId].admin == nodeOperator.admin &&
+        keccak256(abi.encode(s_nodeOperators[nodeOperatorId].name)) == keccak256(abi.encode(nodeOperator.name))
+      ) {
+        revert InvalidNodeOperatorUpdate();
+      }
+      s_nodeOperators[nodeOperatorId].admin = nodeOperator.admin;
+      s_nodeOperators[nodeOperatorId].name = nodeOperator.name;
+      emit NodeOperatorUpdated(nodeOperatorId, nodeOperator.admin, nodeOperator.name);
+    }
```

Comment:
```suggestion
      if (
        s_nodeOperators[nodeOperatorId].admin != nodeOperator.admin ||
        keccak256(abi.encode(s_nodeOperators[nodeOperatorId].name)) != keccak256(abi.encode(nodeOperator.name))
      ) {
         s_nodeOperators[nodeOperatorId].admin = nodeOperator.admin;
         s_nodeOperators[nodeOperatorId].name = nodeOperator.name;
         emit NodeOperatorUpdated(nodeOperatorId, nodeOperator.admin, nodeOperator.name);
      }
    }
```

This would be more efficient in the happy path, right?

Reply:
I think revert is unnecessary - we can make this OP idempotent